content/Watsonx/WatsonxAI/204.md

```diff
@@ -0,0 +1,138 @@
+
+To summarize, we should follow these steps when implementing the RAG use case:
+
+1. Select a vector database. When working with watsonx.ai, we should confirm that the selected vector database is supported. [Check IBM documentation for the latest updates](https://dataplatform.cloud.ibm.com/docs/content/wsj/analyze-data/fm-rag.html?context=wx&audience=wdp).
+2. Select the embeddings model. Confirm that the embeddings model works with the selected vector database.
+3. Experiment with chunk size.
+
+As we discussed earlier, the second part of the RAG use case, sending instructions to LLM, is similar to non-RAG patterns. Developers should pay special attention to token limit because retrieved content will be added to the prompt. If the token limit is exceeded on input, then the model will not be able to generate ouput. This check will need to be implemented as "custom code" in your LLM applications.
+
+WML API does return errors if token size is exceeded.
+
+![wml_error](./images/204/4.png)
+
+## RAG on documents with LangChain and ChromaDB
+
+To run the lab for this section we will start by logging into the watsonx platform; after navigating to the watsonx home page [here](https://dataplatform.cloud.ibm.com/wx/home), we will want to open the Notebook editor that we can use to run the notebook associated with this lab.
+
+> If you don't know how to acccess watsonx.ai or are unsure how to open to the notebook editor, follow [this reference link](/watsonx/watsonxai/100#how-do-i-import-a-jupyter-notebook-in-watsonxai) which will walk you through the process for accessing watsonx.ai and opening the Jupyter notebook editor.
+
+Use the following values for this lab:
+
+- **Name:** `{uniqueid}-rag-chromadb`
+- **Notebook URL:** `https://raw.githubusercontent.com/ibm-build-lab/VAD-VAR-Workshop/main/content/Watsonx/WatsonxAI/files/204/use_case_rag.ipynb`
+
+This lab also requires some local files to source content from. These are located here:
+
+- **Generative_AI_Overview.pdf**: `https://raw.githubusercontent.com/ibm-build-lab/VAD-VAR-Workshop/main/content/Watsonx/WatsonxAI/files/204/Generative_AI_Overview.pdf`
+- **state_of_the_union.txt**: `https://raw.githubusercontent.com/ibm-build-lab/VAD-VAR-Workshop/main/content/Watsonx/WatsonxAI/files/204/state_of_the_union.txt`
+
+You will need to import these as assets into you watsonx.ai instance in order to use them within the notebook.
+
+After your notebook is launched and created, you can follow along and run through each cell of the notebook to complete the lab. The notebook contains comments explaining what code in each cell does as well as any necessary input that you might need to provide in order to successfully run a cell.
+
+Good luck!
+
+## RAG using a website (optional)
+
+This lab is very similar to the last lab. However, unlike the previous instead of sourcing the content from a given file you will instead implement a web scraper that allows a question to be answered about a given webpage.
```

Comment:
I am not sure what `the last lab` is referring to here since this is the only L4 lab in the PR. You might want to link to which lab you are referring to

Reply:
It means to reference the last notebook so I changed it so say `last notebook`.